src/main/java/net/sf/log4jdbc/StatementSpy.java

```diff
@@ -25,15 +25,34 @@
 import java.util.List;
 import java.util.ArrayList;
 
+import static net.sf.log4jdbc.Utilities.getConfigProp;
+
 /**
  * Wraps a Statement and reports method calls, returns and exceptions.
  *
  * jdbc 4 version
  *
  * @author Arthur Blake
  */
-public class StatementSpy implements Statement, Spy
-{
+public class StatementSpy implements Statement, Spy {
+  static final String SQL_SHOW_PARAMS_KEY = "LOG4JDBC_SQL_SHOW_PARAMS";
+
+  static final boolean showParams;
+
+  static {
+    showParams = getConfigProp(SQL_SHOW_PARAMS_KEY, v -> {
+      switch (v.trim().toLowerCase()) {
+        case "true":
+          return true;
+        case "false":
+          return false;
+        default:
+          throw new IllegalArgumentException(String.format(

```

Comment:
What do you think of logging an error or warning and considering it false? instead of throwing exception.

Reply:
I'm OK with both options, but from my opinion exception is better because in case of misconfiguration an app won't start. PII is quite sensitive, so I see throwing exception as a kind of a safety net.

Reply:
I hear you ... generally speaking I would do it that way too. But the default flag value we are choosing is the safe one (not show sensitive info), so I am inclined not to throw exception but leave a big banner log message about it if configured with incorrect value. Especially, it is critical not to abort startup on managed envs (when QA is on fire 🔥 )